src/components/PaginatedTable/utils.tsx

```diff
@@ -23,3 +25,17 @@ export function rafThrottle<Fn extends (...args: any[]) => any>(fn: Fn) {
 export function calculateColumnWidth(newWidth: number, minWidth = 40, maxWidth = Infinity) {
     return Math.max(minWidth, Math.min(newWidth, maxWidth));
 }
+
+export const typedMemo: <T>(Component: T) => T = React.memo;
+
+export function calculateElementOffsetTop(element: HTMLElement, container?: HTMLElement): number {
+    let currentElement = element;
+    let offsetTop = 0;
+
+    while (currentElement && currentElement !== container) {
+        offsetTop += currentElement.offsetTop;
+        currentElement = currentElement.offsetParent as HTMLElement;
```

Comment:
container is container that is being scrolled
element is table element of enormous height

if there are some elements (like group container) between element and container - the offsetTop of element will be offset relative to intermediate element, so we need to take into account parent's offset


scrollContainer -----> group container -----> table

if you get table offsetTop in this schema - it will be distance between table top edge and group container bottom edge while you need distance between scrollContainer and table

Reply:
done